requests/utils.py

```diff
@@ -868,8 +868,8 @@ def check_header_validity(header):
         if not pat.match(value):
             raise InvalidHeader("Invalid return character or leading space in header: %s" % name)
     except TypeError:
-        raise InvalidHeader("Header value %s must be of type str or bytes, "
-                            "not %s" % (value, type(value)))
+        raise InvalidHeader("Header %s value %s must be of type str or bytes, "
```

Comment:
This may read a little clear if the name and/or value are wrapped in quotes.

Perhaps even adjusting the wording to something like "Header {name}: {value} must have a value of string or bytes, not {type}".